### DISCUSSION TITLE
Powerline theme not displaying conda environment

### DISCUSSION THREAD

**Discussion by awa5114** [JanDeDobbeleer/oh-my-posh#1331] [Troubleshoot] [answered]
I am using Windows Terminal in combination with oh-my-posh to customize my command line. I created a new profile in the Windows Terminal `settings.json` file as follows:

```
            {
                "commandline": "powershell.exe -ExecutionPolicy ByPass -NoExit -Command '& C:\\Users\\AA\\Miniconda3\\shell\\condabin\\conda-hook.ps1 ; conda activate C:\\Users\\AA\\Miniconda3'",
                "guid": "{75549856-2f4b-4e71-842a-48288c8f7250}",
                "hidden": false,
                "name": "Anaconda powershell prompt",
                "font": {
                  "size": 10
                }
            }
```

The `commandline` parameter basically contains the command used to generate Anaconda Powershell Prompt, available by default when installing Anaconda.

In my `$profile`, I have the following:

```
Import-Module oh-my-posh
Set-PoshPrompt -Theme powerline 
```

My understanding is that the `powerline` theme has a `python` segment, and `python` segments support Anaconda as mentioned in the [docs](https://ohmyposh.dev/docs/python).

However, when I fire up a new terminal using the `Anaconda powershell prompt` profile in Windows Terminal, the `powerline` theme does not display the base Anaconda environment:

![image](https://user-images.githubusercontent.com/24267968/143674169-6c751353-4334-413c-8940-f07db98a5b95.png)

I know, however that the Anaconda environment is there, because when I do `conda info --envs` it gives me output:

```
# conda environments:
#
base                  *  C:\Users\AA\Miniconda3
temp                     C:\Users\AA\Miniconda3\envs\temp
```

Why does `powerline` not display my environment? Am I doing anything wrong here?

**Comment by JanDeDobbeleer** [comment, marked as answer]
By default base is hidden (as that's not a specific env, but the default Python runtime on your machine).

**Reply by awa5114** [reply]
I tried activating an environment other than `base`. I still can't see it:

```
username@machine_id  C:\Windows\system32  conda activate temp
username@machine_id  C:\Windows\system32  conda info --envs
# conda environments:
#
base                     C:\Users\username\Miniconda3
temp                  *  C:\Users\username\Miniconda3\envs\temp
```

**Reply by JanDeDobbeleer** [reply]
Can you run `Write-PoshDebug`?

**Reply by awa5114** [reply]
Here's what I get with `write-poshdebug`:
```
Here are the timings of segments in your prompt:

ConsoleTitle(false) -   0 ms -
root(false)    -   1 ms -
session(true)  -   0 ms -  921479@L17791
path(true)     -   0 ms -  C:\Windows\system32 
git(false)     -  60 ms -
python(false)  -  14 ms -
exit(true)     -   0 ms - 1337 
```

On another note, I noticed that when I `cd` to a folder which contains `.py` files the environment name *does* appear in my prompt, including if it is the base environment.

I still think there is a problem though, because I'm not always necessarily running the prompt in a directory which contains `.py` files.

**Reply by JanDeDobbeleer** [reply]
Can you update oh-my-posh first? That's not the latest version for sure.

**Reply by awa5114** [reply]
I updated it with `update-module oh-my-posh`. It seems to work as expected now. Thanks a lot for your support!
